data/battle_scripts_1.s

```diff
@@ -8339,6 +8501,7 @@ BattleScript_PrimordialSeaActivates::
 	waitstate
 	playanimation BS_BATTLER_0, B_ANIM_RAIN_CONTINUES
 	call BattleScript_WeatherFormChanges
+	call BattleScript_ActivateWeatherAbilities
```

Comment:
you can simplify all of these calls by putting this loop in sequence after `BattleScript_WeatherFormChanges` has finished. Or better yet, combine weather change abilities into the new `ABILITYEFFECT_ON_WEATHER` case

Reply:
> you can simplify all of these calls by putting this loop in sequence after `BattleScript_WeatherFormChanges` has finished.

As in calling `activateweatherabilities sBATTLER` directly from within `BattleScript_WeatherFormChangesLoop`? :eyes:

I didn't think about it. That sounds nice to save on loops :eyes:

> Or better yet, combine weather change abilities into the new `ABILITYEFFECT_ON_WEATHER` case

Gah. That sounds more involved, but also sounds like the way to go. I'll start to look into it and see what I can do :eyes: